uaclient/files.py

```diff
@@ -0,0 +1,344 @@
+import copy
+import json
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Optional
+
+from uaclient import (
+    defaults,
+    event_logger,
+    exceptions,
+    messages,
+    schemas,
+    util,
+)
+from uaclient.machine_token_data import PublicMachineTokenData
+
+event = event_logger.get_event_logger()
+LOG = logging.getLogger(__name__)
+
+
+class UAFile:
+    def __init__(
+        self,
+        name: str,
+        root_directory: str = defaults.DEFAULT_DATA_DIR,
```

Comment:
Bikeshedding: "root_directory" has a specific meaning which is `/`.
Can we call this just `directory`?

Reply:
Yeah, `directory` is a better name.